src/server/game/Server/WorldSession.cpp

```diff
@@ -748,6 +748,46 @@ void WorldSession::LoadAccountData(PreparedQueryResult result, uint32 mask)
     while (result->NextRow());
 }
 
+void WorldSession::LoadAccountToys(PreparedQueryResult result)
+{
+    if (!result)
+        return;
+
+    do
+    {
+        Field* fields = result->Fetch();
+        uint32 itemId = fields[0].GetUInt32();
+        bool isFavourite = fields[1].GetBool();
+
+        _toys[itemId] = isFavourite;
+    }
+    while (result->NextRow());
+}
+
+void WorldSession::SaveAccountToys(SQLTransaction& trans)
+{
+    PreparedStatement* stmt = NULL;
+    for (ToyBoxContainer::const_iterator itr = _toys.begin(); itr != _toys.end(); ++itr)
+    {
+        stmt = LoginDatabase.GetPreparedStatement(LOGIN_REP_ACCOUNT_TOYS);
+        stmt->setUInt32(0, GetBattlenetAccountId());
+        stmt->setUInt32(1, itr->first);
+        stmt->setBool(2, itr->second);
+        trans->Append(stmt);
+    }
+}
+
+bool WorldSession::UpdateAccountToys(uint32 itemId, bool isFavourite /*= false*/)
+{
+    ToyBoxContainer::const_iterator itr = _toys.find(itemId);
+    if (itr != _toys.end())
+        return false;
+
+    _toys[itemId] = isFavourite;
```

Comment:
since there is already a find() operation above, could the itr be used instead of calling the [ ] overload here ?